### ISSUE TITLE
Too long filename causes bug

### ISSUE THREAD

**Issue by barbafa** [zim-desktop-wiki/zim-desktop-wiki#570]
When trying to make a new note that if created would make the total filename including folders too long for win10.
Steps to reproduce: 

1. File system for harddrive is NTFS
2. Folder is under Dropbox (b/c syncing)
3. Create nested notes to a total depth of ca 250(?) chars
4. Create a new note with a filename longre than 10 chars
5. Zim Desktop Wiki gives a pop-up window with an error message:
6. "[Errno 2] No such file or directory: " plus the erronous file/foldername
7. In the sidepane with the filetree, the new filename that I tried to create is greyed out.
8. The GUI were still running at this point, so I tried to just switch back to a higher level. 
9. That was not possible - when clicking in the sidepane on the higher level I got no reaction from Zim.
10. So I tried clicking on the Home button in the toolbar. (Using alt-left gives the same error message.) Then Zim gave me a new error message:

```
Looks like you found a bug
This is zim 0.68
Platform: nt
Locale: sv_SE cp1252
FS encoding: mbcs
Python: (2, 7, 14, 'final', 0)
Gtk: (2, 24, 10)
Pygtk: (2, 24, 0)
No bzr version-info found

======= Traceback =======
  File "zim\actions.pyo", line 157, in do_activate
  File "zim\actions.pyo", line 140, in func
  File "zim\gui\__init__.pyo", line 874, in open_page_home
  File "zim\actions.pyo", line 140, in func
  File "zim\gui\__init__.pyo", line 730, in open_page
AssertionError: Could not close page
```
So basically Zim refuses to switch from the note that is too long for Windows to handle, probably when trying to save the note.

The easy workaround is to simply in the notes text window select all text and erase it. Then Zim erases the note and wont try to save it when one is exiting that note.

It would be preferable if Zim handled it otherwise though. I hope the file system gives an error message when Zim tries to create the too long filename, and that error message can be captured by Zim? Then Zim could either auto-shorten the filename, or give an informative error message to the user.

Regards, Barbafa

**Comment by jaap-karssenberg** [comment]
Agree, should at least give (human readable) error when trying to open such page.

**Comment by jaap-karssenberg** [comment]
Dev notes:
  * To reproduce it is crucial to close the page by navigating away
  * "discard changes" does work here
  * Cancel leads to "Found bug - could not save page"
  * Recover from cancel by cancelling "open_page" action (instead of AssertionError line 753)
  * Test this scenario, ensure pageview gives the dialog, so cancel applies to open_page as well
